src/vm/vm_core.rs

```diff
@@ -422,6 +427,15 @@ impl VirtualMachine {
     }
 
     pub fn step(&mut self) -> Result<(), VirtualMachineError> {
+        if let Some(hint_list) = self.hints.get(&self.run_context.pc) {
+            for hint_code in hint_list.clone().iter() {
```

Comment:
Check if there is any way not to clone the hint list. I will accept it for now but please create an issue with a link to this line so that we remove it afterwards